packages/core/src/indexing/profile.ts

```diff
@@ -0,0 +1,675 @@
+import type { Event } from "@/internal/types.js";
+import { orderObject } from "@/utils/order.js";
+import type { Abi } from "viem";
+import {
+  type PonderActions,
+  type ProfilePattern,
+  type Request,
+  encodeRequest,
+  getCacheKey,
+} from "./client.js";
+
+export const getProfilePatternKey = (pattern: ProfilePattern): string => {
+  return JSON.stringify(orderObject(pattern), (key, value) => {
+    if (key === "abi") return undefined;
+
+    if (typeof value === "bigint") {
+      return value.toString();
+    }
+    return value;
+  });
+};
```

Comment:
@typedarray Can you take a look at this? 

Is this a reasonable way to get a unique key from the pattern?